client/src/main/java/io/split/service/SplitHttpClientImpl.java

```diff
@@ -0,0 +1,113 @@
+    private final CloseableHttpClient _client;
+    private final RequestDecorator _requestDecorator;
+
+    public static SplitHttpClientImpl create(
+            CloseableHttpClient client,
+            RequestDecorator requestDecorator
+    ) throws URISyntaxException {
+        return new SplitHttpClientImpl(client, requestDecorator);
+    }
+
+    private SplitHttpClientImpl
+            (CloseableHttpClient client,
+             RequestDecorator requestDecorator) {
+        _client = client;
+        _requestDecorator = requestDecorator;
+    }
+
+    public SplitHttpResponse get(URI uri, FetchOptions options) {
+        CloseableHttpResponse response = null;
+
+        try {
+            HttpGet request = new HttpGet(uri);
+            if(options.cacheControlHeadersEnabled()) {
+                request.setHeader(HEADER_CACHE_CONTROL_NAME, HEADER_CACHE_CONTROL_VALUE);
+            }
+            request = (HttpGet) _requestDecorator.decorateHeaders(request);
+
+            response = _client.execute(request);
+
+            int statusCode = response.getCode();
+
+            if (_log.isDebugEnabled()) {
+                _log.debug(String.format("[%s] %s. Status code: %s", request.getMethod(), uri.toURL(), statusCode));
+            }
+
+            SplitHttpResponse httpResponse = new SplitHttpResponse();
+            httpResponse.statusMessage = "";
+            if (statusCode < HttpStatus.SC_OK || statusCode >= HttpStatus.SC_MULTIPLE_CHOICES) {
+                _log.warn(String.format("Response status was: %s. Reason: %s", statusCode , response.getReasonPhrase()));
+                httpResponse.statusMessage = response.getReasonPhrase();
+            }
+            httpResponse.statusCode = statusCode;
+            httpResponse.body = EntityUtils.toString(response.getEntity(), StandardCharsets.UTF_8);
+            return httpResponse;
+        } catch (Exception e) {
+            throw new IllegalStateException(String.format("Problem in http get operation: %s", e), e);
+        } finally {
+            Utils.forceClose(response);
+        }
+    }
+
+    public SplitHttpResponse post
+            (URI uri,
+             HttpEntity entity,
+             Map<String, String> additionalHeaders) throws IOException {
+        CloseableHttpResponse response = null;
+        long initTime = System.currentTimeMillis();
+        try {
+            HttpPost request = new HttpPost(uri);
+            if (additionalHeaders != null) {
+                for (Map.Entry entry : additionalHeaders.entrySet()) {
+                    request.addHeader(entry.getKey().toString(), entry.getValue());
+                }
+            }
+            request.setEntity(entity);
+            request = (HttpPost) _requestDecorator.decorateHeaders(request);
+
+            response = _client.execute(request);
+
+            int status = response.getCode();
+
+            String statusMessage = new String("");
```

Comment:
```suggestion
            String statusMessage;
```

Reply:
If I remove this, I get error on line 104: statusMessage might not be initialized
httpResponse.statusMessage = statusMessage;

Reply:
will do